homeassistant/components/sensor/nextbus.py

```diff
@@ -0,0 +1,271 @@
+    """Sensor class that displays upcoming NextBus times.
+
+    To function, this requires knowing the agency tag as well as the tags for
+    both the route and the stop.
+
+    This is possibly a little convoluted to provide as it requires making a
+    request to the service to get these values. Perhaps it can be simplifed in
+    the future using fuzzy logic and matching.
+    """
+
+    def __init__(self, agency, route, stop, name=None):
+        """Initialize sensor with all required config."""
+        self.agency = agency
+        self.route = route
+        self.stop = stop
+        self._custom_name = name
+        self._name = '{} - {}'.format(agency, route)
+
+        # Initialize a unique logger
+        self._logger = _LOGGER.getChild('.'.join((
+            self.__class__.__name__,
+            self.agency,
+            self.route,
+            self.stop,
+        )))
+
+        # set up default state attributes
+        self._state = 0
+        self._attributes = {}
+
+    @property
+    def name(self):
+        """Return sensor name.
+
+        Uses an auto generated name based on the data from the API unless a
+        custom name is provided in the configuration.
+        """
+        if self._custom_name:
+            return self._custom_name
+
+        return self._name
+
+    @property
+    def state(self):
+        """Return current state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return ATTR_MINUTES
+
+    @property
+    def device_state_attributes(self):
+        """Return additional state attributes."""
+        return self._attributes
+
+    @property
+    def icon(self):
+        """Return icon to be used for this sensor."""
+        # Would be nice if we could determine if the line is a train or bus
+        # however that doesn't seem to be available to us. Using bus for now.
+        return ICON
+
+    def update(self):
+        """Update sensor with new departures times."""
+        # Note: using Multi because there is a bug with the single stop impl
+        predictions = get_predictions(
+            [{
+                'stop_tag': int(self.stop),
+                'route_tag': self.route,
+            }],
+            self.agency,
+        )
+
+        self._logger.debug('Predictions results: %s', predictions)
+
+        if 'Error' in predictions:
+            self._logger.error('Could not get predictions: %s', predictions)
+
+        if not predictions.get('predictions'):
+            self._logger.info('No predictions available')
+            self._state = None
+            # Remove attributes that may now be outdated
+            self._attributes.pop('Upcoming', None)
+            return
+
+        predictions = predictions['predictions']
+
+        # Set detailed attributes
+        self._attributes.update({
+            'Agency': predictions.get('agencyTitle'),
```

Comment:
State attribute keys should be lowercase snakecase.

Reply:
Is there a way to display a label for each attribute so that the UI is user friendly? When  user clicks on the sensor, it would be a bad experience to see snake case.

Reply:
No.